sys/vm_map.c

```diff
@@ -245,9 +241,8 @@ vm_map_t *vm_map_clone(vm_map_t *map) {
 
   rw_scoped_enter(&map->rwlock, RW_READER);
 
-  /* Temporarily switch to the new map, so that we may write contents. Note that
-     it's okay if we get preempted - the working vm map will be restored on
-     context switch. */
+  /* Temporarily switch to the new map, so that we may write contents. */
+  critical_enter();
```

Comment:
We'd need to check if there is only one thread in this process *AND* set `proc->p_uspace = newmap` for the duration of the copy process.